types/dialogify-browser/dialogify-browser-tests.ts

```diff
@@ -0,0 +1,279 @@
+// $ExpectAssignable
+const dialogStyle: Dialogify.DialogStyle = {
+    style: {},
+    className: '',
+    contentStyle: {},
+    contentClassName: '',
+};
+// $ExpectAssignable
+const buttonStyle: Dialogify.ButtonStyle = {
+    image: '',
+    style: {},
+    className: '',
+};
+// $ExpectAssignable
+const dialogOptions: Dialogify.DialogOptions = {
+    size: Dialogify.SIZE_LARGE,
+    closable: true,
+    fixed: true,
+    dialog: dialogStyle,
+    closeButton: buttonStyle,
+    useDialogForm: true,
+    ajaxPrefix: '/',
+    ajaxData: '',
+    ajaxComplete: () => void 0,
+};
+
+// $ExpectError
+new Dialogify();
+// $ExpectError
+new Dialogify('', '');
+// $ExpectType Dialogify
+const dialog = new Dialogify('source', dialogOptions);
+
+// $ExpectType JQuery<HTMLElement>
+dialog.$content;
+// $ExpectType { [key: string]: ButtonImpl; [key: number]: ButtonImpl; }
+dialog.$buttonList;
+
+// $ExpectError
+dialog.title();
+// $ExpectError
+dialog.title(0);
+// $ExpectType Dialogify
+dialog.title('');
+// $ExpectError
+dialog.title('', '');
+
+// $ExpectAssignable
+const buttons: Array<Dialogify.Button | string> = [
+    {
+        type: '',
+        text: '',
+        click: () => void 0,
+        focused: true,
+        disabled: false,
+        id: '',
+    },
+    '',
+];
+
+// $ExpectAssignable
+const options: Dialogify.ButtonOption = {
+    position: '',
+};
+
+// $ExpectError
+dialog.buttons();
+// $ExpectError
+dialog.buttons('', '');
+// $ExpectType Dialogify
+dialog.buttons(buttons);
+// $ExpectType Dialogify
+dialog.buttons(buttons, options);
+// $ExpectError
+dialog.buttons(buttons, options, '');
+
+// $ExpectError
+dialog.on();
+// $ExpectError
+dialog.on('', () => void 0);
+// $ExpectType Dialogify
+dialog.on('show', () => void 0);
+// $ExpectType Dialogify
+dialog.on('close', () => void 0);
+// $ExpectType Dialogify
+dialog.on('cancel', () => void 0);
```

Comment:
```suggestion
dialog.on('', () => {});
// $ExpectType Dialogify
dialog.on('show', () => {});
// $ExpectType Dialogify
dialog.on('close', () => {});
// $ExpectType Dialogify
dialog.on('cancel', () => {});
```
not exactly the common way, avoid this.

Reply:
Fixed!